install.sh

```diff
@@ -85,69 +85,49 @@ fi
-  # The --retry/--retry-all-errors parameter in curl will not work with 
+  # The --retry/--retry-all-errors parameter in curl will not work with
   # the 'curl: (7) Couldn't connect to server' error, a for loop is used
   # here.
   for (( i = 0; i < 4; i++ )); do
-    env LD_LIBRARY_PATH='' ${CURL} --ssl -C - "${@}" && \
-      return 0 || \
-      echo_info "Retrying, $((3-$i)) retries left."
+    env LD_LIBRARY_PATH=''
+    if [ "$("${CURL}" --ssl -C - "${@}")" ]; then
+      return 0
+    else
+      echo_info "Retrying, $((3-i)) retries left."
+    fi
   done
   # The download failed if we're still here.
   echo_error "Download failed :/ Please check your network settings."
   return 1
 }
 
-case "${ARCH}" in
-"i686"|"x86_64"|"armv7l"|"aarch64")
-  LIB_SUFFIX=
-  [ "${ARCH}" == "x86_64" ] && LIB_SUFFIX='64'
-  ;;
-*)
-  echo_error "Your device is not supported by Chromebrew yet :/"
-  exit 1;;
-esac
-
 echo_info "\n\nDoing initial setup for install in ${CREW_PREFIX}."
-echo_info "This may take a while if there are preexisting files in ${CREW_PREFIX}...\n"
-
-# This will allow things to work without sudo.
-crew_folders="bin cache doc docbook etc include lib lib$LIB_SUFFIX libexec man sbin share tmp var"
-for folder in $crew_folders
-do
-  if [ -d "${CREW_PREFIX}"/"${folder}" ]; then
-    echo_intra "Resetting ownership of ${CREW_PREFIX}/${folder}"
-    sudo chown -R "$(id -u)":"$(id -g)" "${CREW_PREFIX}"/"${folder}"
-  fi
-done
-sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
 
-# Delete ${CREW_PREFIX}/{var,local} symlinks on some Chromium OS distro 
-# if they exist.
+# Delete ${CREW_PREFIX}/{var,local} symlinks which exist on some Chromium OS distros.
 [ -L ${CREW_PREFIX}/var ] && sudo rm -f "${CREW_PREFIX}/var"
 [ -L ${CREW_PREFIX}/local ] && sudo rm -f "${CREW_PREFIX}/local"
 
-# Prepare directories.
-for dir in "${CREW_CONFIG_PATH}/meta" "${CREW_DEST_DIR}" "${CREW_PACKAGES_PATH}" "${CREW_CACHE_DIR}" ; do
-  if [ ! -d "${dir}" ]; then
-    mkdir -p "${dir}"
-  fi
-done
-
-echo_info "\nDownloading information for Bootstrap packages..."
-echo -en "${GRAY}"
-# Use parallel mode if available.
-if [[ "$(curl --help curl)" =~ --parallel ]]; then
-  (cd "${CREW_LIB_PATH}"/packages && curl -OLZ "${URL}"/packages/{"${BOOTSTRAP_PACKAGES// /,}"}.rb)
+if [ "$(ls -A ${CREW_PREFIX})" ]; then
+  echo_error "${CREW_PREFIX} is not empty, please move all files out of ${CREW_PREFIX}."
+  exit 1
 else
-  (cd "${CREW_LIB_PATH}"/packages && curl -OL "${URL}"/packages/{"${BOOTSTRAP_PACKAGES// /,}"}.rb)
+  sudo rm -rf ${CREW_PREFIX}
 fi
-echo -e "${RESET}"
 
-# Prepare url and sha256.
+# This will create the directories
+crew_folders="bin cache doc docbook include lib/crew/packages lib$LIB_SUFFIX libexec man sbin share var etc/crew/meta tmp/crew/dest"
+for folder in $crew_folders; do
+  sudo mkdir -p "${CREW_PREFIX}"/"${folder}"
+done
+
+# This will allow things to work without sudo
+sudo chown -R "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
```

Comment:
Not needed if the commands run above are without sudo.

Reply:
As in lines 123 and 124 are not needed?

Reply:
Yes, the chown line is not needed.